### DISCUSSION TITLE
Unable to use util in wix3

### DISCUSSION THREAD

**Discussion by qrnch-jan** [wixtoolset/issues#6892] [Questions] [answered]
Hello,

I'm trying to register an event source from an installer, but I get the error:

```
error CNDL0200 : The Component element contains an unhandled extension element 'util:EventSource'.  Please ensure that the extension for elements in the 'http://schemas.microsoft.com/wix/UtilExtension' namespace has been provided.
```

What I have done:
- The `.wsx` file contains, on the `Wix` tag, `xmlns:util="http://schemas.microsoft.com/wix/UtilExtension"`
- The `light.exe` command line has the arguments `-ext WixUtilExtension`
- The tag is referenced as `<util:EventSource>`

What I've tried:
- I tried to use another `util` component (`ServiceConfig`), but I get the same error (well .. but for `util:ServiceConfig`), so I get the distinct feeling that this problem is confined to `util`.  (I am able to use UI extensions (with the appropriate `-ext`)).
- I've cut'n'paste from examples which I presume work, and I get the same problem -- making me think there's something wrong with my installation.  (The `WixUtilExtension.dll` is in the `bin` directory).

I'm using version 3.11.2 on Windows 11.

Living by the mantra "How hard can it be?" I chose to ignore all the "Don't try to register/deregister your service manually from within the installer" warnings I stumbled on and had the installer run the service executable (because it can register the service and register the event source) as a custom action.  I believe I aged roughly three years during the two days I was working on this.  (Learned a lot though, but the most important lesson was to heed warnings about trying to avoid using built-in installer features).

**Comment by robmen** [comment, marked as answer]
Did you include the `-ext` switch to `candle.exe`? `CNDL0200` is a compiler (`candle.exe`) error.

**Reply by qrnch-jan** [reply]
Sigh..  I can't believe I didn't notice that.  Another lesson learned:  It's not always a good idea to stick things in a script.  I would have caught this if I had run candle manually.

Thank you!
